### PR TITLE
Add Radio module

### PR DESCRIPTION
**Pull request by diogo-mendes**
In this PR we add the Radio module to the Y design, which includes the `Radio` (button) and `RadioGroup` components.

Please check out the storybook for a complete description of the defined API and extendable styles.

### CODE REVIEW

File: src/modules/Radio/Radio.tsx [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+import {
+  StandardProps,
+  WithStylesProps,
+  ThemeStyleSheetFactory
+} from "../../types";
+import { withStyles } from "../../base";
+
+interface IProps extends StandardProps<"input"> {
+  hasError?: boolean;
+  isDisabled?: boolean;
```
Comment (nunoliveira12) [inline src/modules/Radio/Radio.tsx]: you should use "disabled" , because it is already passed as a prop for the input

Comment (nunoliveira12) [reply src/modules/Radio/Radio.tsx]: so you can remove it from IProps

File: src/modules/Radio/Radio.tsx [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+import uuid from "uuid/v4";
+import {
+  StandardProps,
+  WithStylesProps,
+  ThemeStyleSheetFactory
+} from "../../types";
+import { withStyles } from "../../base";
+
+interface IProps extends StandardProps<"input"> {
+  hasError?: boolean;
```
Comment (nunoliveira12) [inline src/modules/Radio/Radio.tsx]: on other components is only "error"

File: src/modules/Radio/Radio.tsx [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+  isDisabled?: boolean;
+  label: string;
+  name?: string;
+  selectedValue?: string;
+  value: string;
```
Comment (nunoliveira12) [inline src/modules/Radio/Radio.tsx]: same as before

File: src/modules/Radio/Radio.tsx [hunk 4]
```diff
@@ -0,0 +1,123 @@
...
+  onValueSelect?: (value: string) => void;
```
Comment (nunoliveira12) [inline src/modules/Radio/Radio.tsx]: should use onChange of input

File: src/modules/Radio/Radio.tsx [hunk 5]
```diff
@@ -0,0 +1,123 @@
...
+      "&.checked": {},
+      "&.disabled": {},
+      "&.error": {
+        color: theme.colors.alert.alert50
+      }
+    }
+  }
+});
+
+class Radio extends React.PureComponent<IProps & WithStylesProps> {
```
Comment (nunoliveira12) [inline src/modules/Radio/Radio.tsx]: should be function component

File: src/modules/Radio/Radio.tsx [hunk 6]
```diff
@@ -0,0 +1,123 @@
...
+  WithStylesProps,
+  ThemeStyleSheetFactory
+} from "../../types";
+import { withStyles } from "../../base";
+
+interface IProps extends StandardProps<"input"> {
+  hasError?: boolean;
+  isDisabled?: boolean;
+  label: string;
+  name?: string;
```
Comment (nunoliveira12) [inline src/modules/Radio/Radio.tsx]: same

File: src/modules/Radio/Radio.tsx [hunk 7]
```diff
@@ -0,0 +1,123 @@
...
+            isDisabled && "disabled",
+            hasError && "error"
+          )}
+          id={id}
+          name={name}
+          ref={wrappedRef}
+          type="radio"
+          value={value}
+          onChange={this.handleChange.bind(this)}
+        />
```
Comment (nunoliveira12) [inline src/modules/Radio/Radio.tsx]: you can use {...props} after "removing" all the custom props on the beginning like (label, ...props)

File: src/modules/Radio/Radio.tsx [hunk 8]
```diff
@@ -0,0 +1,123 @@
...
+      "&.error": {
+        color: theme.colors.alert.alert50
+      }
+    }
+  }
+});
+
+class Radio extends React.PureComponent<IProps & WithStylesProps> {
+  // eslint-disable-next-line react/state-in-constructor
+  state = {
```
Comment (nunoliveira12) [inline src/modules/Radio/Radio.tsx]: hoooooooks

File: src/modules/Radio/Radio.tsx [hunk 9]
```diff
@@ -0,0 +1,123 @@
...
+  ThemeStyleSheetFactory
+} from "../../types";
+import { withStyles } from "../../base";
+
+interface IProps extends StandardProps<"input"> {
+  hasError?: boolean;
+  isDisabled?: boolean;
+  label: string;
+  name?: string;
+  selectedValue?: string;
```
Comment (nunoliveira12) [inline src/modules/Radio/Radio.tsx]: should avoid this extra prop and have instead a hook to get the parent context of the radioGroup and get its current value

File: src/modules/Radio/RadioGroup.tsx [hunk 10]
```diff
@@ -0,0 +1,37 @@
+import React from "react";
+import { WithStylesProps, ThemeStyleSheetFactory } from "../../types";
+import { withStyles } from "../../base";
+
+interface IProps {
```
Comment (nunoliveira12) [inline src/modules/Radio/RadioGroup.tsx]: extend StardardProps<"input">(?) just to keep the default props like children, styles, classname , onchange, disabled, value

File: src/modules/Radio/Radio.tsx [hunk 11]
```diff
@@ -0,0 +1,123 @@
...
+  value: string;
+  onValueSelect?: (value: string) => void;
+}
+
+const styleSheet: ThemeStyleSheetFactory = theme => ({
+  input: {
+    color: theme.colors.primary,
+    cursor: "pointer",
```
Comment (Henrts) [inline src/modules/Radio/Radio.tsx]: Add "margin": 0 here

File: src/modules/Radio/Radio.tsx (hunk 4 above)
Comment (diogo-mendes) [reply src/modules/Radio/Radio.tsx]: As we discussed irl, I don't think that's what I want!

File: src/modules/Radio/Radio.tsx (hunk 3 above)
Comment (diogo-mendes) [reply src/modules/Radio/Radio.tsx]: We need to force the type bc input's "value" can be undefined :(

File: src/modules/Radio/Radio.tsx (hunk 11 above)
Comment (diogo-mendes) [reply src/modules/Radio/Radio.tsx]: As we have seen irl I don't think that's necessary! Let me know if you find that issue again

File: src/modules/Radio/RadioGroup.tsx (hunk 10 above)
Comment (acaldas) [reply src/modules/Radio/RadioGroup.tsx]: bump

**Pull request merged**